File: G_Projects/G_Inventory_Management_System/ProductReplenishmentStrategies/ConcreteReplenishmentStrategies/JustInTimeStrategy.java
package G_Projects.G_Inventory_Management_System.ProductReplenishmentStrategies.ConcreteReplenishmentStrategies;

import G_Projects.G_Inventory_Management_System.ProductFactory.Product;
import G_Projects.G_Inventory_Management_System.ProductReplenishmentStrategies.ReplenishmentStrategy;

public class JustInTimeStrategy implements ReplenishmentStrategy {
    @Override
    public void replenish(Product product){
        System.out.println("Replenishing product " + product.getName() + " using Just-In-Time Strategy.");
    }
}

File: Python/A_Project/D_Inventory_Management_System/ProductReplenishmentStrategies/ConcreteReplenishmentStrategies/just_in_time_strategy.py
from ProductReplenishmentStrategies.replenishment_strategy import ReplenishmentStrategy


class JustInTimeStrategy(ReplenishmentStrategy):
    def replenish(self, product):
        current_quantity = product.get_quantity()
        threshold = product.get_threshold()
        if current_quantity < threshold:
            product.set_quantity(threshold)
        print(
            f"Replenishing product {product.get_name()} using Just-In-Time Strategy."
        )
